WordPress/Classes/ViewRelated/Post/PublishSettingsViewController.swift

```diff
@@ -0,0 +1,179 @@
+import Foundation
+import CocoaLumberjack
+import WordPressShared
+
+private enum PublishSettingsCell: CaseIterable {
+    case dateTime
+
+    var title: String {
+        switch self {
+        case .dateTime:
+            return "Date and time"
+        }
+    }
+}
+
+struct PublishSettingsViewModel {
+    enum State {
+        case scheduled(Date)
+        case published(Date)
+        case immediately
+    }
+
+    private var state: State
+    let timeZone: OffsetTimeZone?
+
+    let title: String?
+
+    private let post: AbstractPost
+
+    init(post: AbstractPost) {
+        if let dateCreated = post.dateCreated {
+            if post.hasFuturePublishDate() {
+                state = .scheduled(dateCreated)
+            } else {
+                state = .published(dateCreated)
+            }
+        } else {
+            state = .immediately
+        }
+
+        self.post = post
+
+        title = post.postTitle
+
+        if let gmtOffset = post.blog.settings?.gmtOffset {
+            timeZone = OffsetTimeZone(offset: gmtOffset.floatValue)
+        } else {
+            timeZone = nil
+        }
+    }
+
+    fileprivate var cells: [PublishSettingsCell] {
+        switch state {
+        case .published, .immediately:
+            return [PublishSettingsCell.dateTime]
+        case .scheduled:
+            return PublishSettingsCell.allCases
```

Comment:
OK, I admit, I don't quite understand why this is returning `allCases`. Can you explain a bit please?

Reply:
Some of this is from code I had prepared for the Notifications and Calendar cases. Was planning to add cell types for each of those and include them here. So in the scheduled case, this will include the `.notification` and `.calendar` type also. Might be better to be more explicit and just use an array for this:

`[PublishSettingsCell.dateTime, PublishSettingsCell.notifications, PublishSettingsCell.calendar]`

Reply:
Oh! _Now_ I get it. Thanks for the explanation. When you add `notification` and `calendar` it will make more sense; then it will be obvious what it's doing. I'm not sure an explicit array will be needed then, but we'll see when you get to that part.